### ISSUE TITLE
Implement open food facts API

### ISSUE THREAD

**Issue by patzly** [patzly/grocy-android#3]
Only product name could be used, get it from barcode.

**Problem:** names in the database are very long, the line breaks would be ugly in the app.
Most names contain the brand, exact product name and possible variant identification, way too much information for a clean management in Grocy.

**Comment by prohand** [comment]
Hello,
Great enhancement 👋😊

**Comment by patzly** [comment]
From the [**Open Food Facts API Documentation**](https://documenter.getpostman.com/view/8470508/SVtN3Wzy?version=latest)

> **READ requests** allow you to retrieve the nutritional data of a product with a barcode.
> 
> GET `https://[countrycode].openfoodfacts.org/api/v0/product/[barcode].json`
>
> Example:
> GET `https://world.openfoodfacts.org/api/v0/product/737628064502.json`
> 
> **Authentication**
> No authentication is required for READ operations.
> However, you have to add a User-Agent HTTP Header with the name of your app, the version, system and a url (if any), not to be blocked by mistake.
> 
> Example: User-Agent: `NameOfYourApp - Android - Version 1.0 - www.yourappwebsite.com`

User-Agent: `Grocy Android - v1.1.0 - https://github.com/patzly/grocy-android`

**Comment by patzly** [comment]
**Json format** to get product name:

`https://world.openfoodfacts.org/api/v0/product/42100454.json`

User-Agent: `Grocy Android - v1.1.0 - https://github.com/patzly/grocy-android`
```json
{
    "product":{
        "traces_tags":{
            "product_name":"Alnatura Mandelschnitte, Mandel"
        }
    }
}
```

**Comment by patzly** [comment]
Also calories could be used when the calories field is implemented in the product edit page.

**Comment by patzly** [comment]
We'll leave it at the product name for now.

**Comment by ovz93br43v7** [comment]
@patzly Probably is it possible to add also:
https://github.com/openfoodfacts/openbeautyfacts and
https://www.codecheck.info/
?

**Comment by MrJohnZoidberg** [comment]
https://www.codecheck.info/ is not possible because the API is not open – and it is also protected via Google Captcha.
But maybe we can implement https://world.openbeautyfacts.org/, I will open a new issue for that.
